core/src/main/scala/org/bitcoins/core/wallet/fee/FeeUnit.scala

```diff
@@ -45,13 +70,38 @@ case class SatoshisPerByte(currencyUnit: CurrencyUnit) extends BitcoinFeeUnit {
     SatoshisPerKiloByte(currencyUnit.satoshis * Satoshis(1000))
   }
 
-  override def txSizeForCalc(tx: Transaction): Long = tx.byteSize
+  override def txSizeForCalc(tx: Transaction): Long =
+    SatoshisPerByte.txSizeForCalc(tx)
 
-  override def scaleFactor: Long = 1
+  override def scaleFactor: Long = SatoshisPerByte.scaleFactor
 }
 
-object SatoshisPerByte {
-  def fromLong(sats: Long): SatoshisPerByte = SatoshisPerByte(Satoshis(sats))
+object SatoshisPerByte extends FeeUnitFactory[SatoshisPerByte] {
+
+  /** The coefficient the denominator in the unit is multiplied by,
+    * for example sats/kilobyte -> 1000
+    */
+  override def scaleFactor: Long = 1
+
+  /** Takes the given transaction returns a size that will be used for calculating the fee rate.
+    * This is generally the denominator in the unit, ie sats/byte
+    */
+  override def txSizeForCalc(tx: Transaction): Long = tx.byteSize
+
+  override def calc(
+      inputAmount: CurrencyUnit,
+      tx: Transaction): SatoshisPerByte = {
+    val feePaid = inputAmount - tx.totalOutput
+    val feeRate = feePaid.satoshis.toLong / txSizeForCalc(tx).toDouble
+
+    fromLong(Math.round(feeRate))
+  }
```

Comment:
why is this here? This seems to be equivalent to the thing it overrides?

Reply:
Woops, missed in my cleanup